selectstar/cloudformation-templates
# RDS: Improve support for firewall-protected RDS instance [sc-28655]

## Description



We are currently updating the firewall rules to allow access from the Select Star platform. However, we also connect to AWS Lambda when provisioning. It's show on diagram:

![image](https://user-images.githubusercontent.com/99484706/184025763-53edd47c-e14e-48d5-af04-42279cc9897d.png)

In that situation, we need to update the firewall rules to enable access from AWS Lambda too. This change primarily changes that before connecting to AWS RDS for PostgreSQL.

It also:

* updates Terraform to add ingress rule late and restrict that rule to local IP address-only
* fix deprecation warning around `aws_s3_bucket.acl`
* fix `smoke.py` by copy dict to avoid password redaction.

## How to reproduce/test?

 

You can use Terraform e2e tests to verify that everything is provisioned successfully, despite removing the ingress rule for "0.0.0.0/0".

<img width="1552" alt="image" src="https://user-images.githubusercontent.com/99484706/184028327-fb1f07e5-8d9f-44d2-bb72-c443ed30d77b.png">


## Type of change


## Related tickets

-   Story sc-28655

## Checklists

    -   [links to related PRs]
